### PR TITLE
feat(environment): can run in containers

### LINKED ISSUE #279: Expand data collection to Docker containers
**Summary**

As stated in the README:

> Telemetry collection runs exclusively in CI environments. Collection will never happen on local development environments or on projects that aren't configured to run automated scripts on a CI environment (GitHub actions, Travis CI, etc.) During a build or any other CI operation that installs package dependencies (npm install, yarn install, ...), IBM telemetry will run as a background process and perform data collection.

The [ci-info](https://github.com/watson/ci-info) package is used to determine if the background process is running in a continuous integration environment or not. This decision was made alongside the collection limit of one scan per repository branch per day, as any more data than this wouldn't be too useful.

We're finding some teams aren't having their Carbon usage properly reported due to their use of Docker and no CI/CD pipeline that runs `npm install`.

**Job stories**

> What value will this deliver?

1. When my project uses Docker and performs an `npm install` inside my Docker container during a manual image build, I want to IBM Telemetry to run, so the Carbon maintaining can learn about my usage.

**Acceptance criteria**

> What requirements must be met to complete this request?

1. Telemetry collection runs when a Docker image is built (if it includes an npm install)
2. README documentation is updated with the new collection logic

**Potential solution**

Consider a package like [is-docker](https://www.npmjs.com/package/is-docker) or something similar to add to the `isCI` check.

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
Closes #279

#### Changelog

**New**

- installed `is-inside-container` package to detect if telemetry is running inside container

#### Testing / reviewing
Added new test case to check if `environment.isCi` is `true` after adding files only found within Docker/Podman container instances

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,6 +46,9 @@
     "test": "vitest run --coverage",
     "test:watch": "vitest --coverage"
   },
+  "dependencies": {
+    "is-inside-container": "^1.0.0"
+  },
```
Comment (mattrosno) [inline package.json:51]: As the only dependency, this seems wrong. I would look into how dependencies are bundled so when this package is installed, it doesn't cause other packages to also be installed.

**Review by mattrosno — CHANGES_REQUESTED** [review]
I think there's an issue with how you installed the new dependency. Also, can you update the readme? With this PR, we need to update:

> Telemetry data collection runs on CI servers, like GitHub Actions, Travis CI, or Jenkins. It never runs locally on a developer's machine.

> Telemetry collection runs exclusively in CI environments. Collection will never happen on local development environments or on projects that aren't configured to run automated scripts on a CI environment (GitHub actions, Travis CI, etc.)

There might be other places in the readme that need updated as well.

File: src/main/core/environment.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 import { isCI } from 'ci-info'
+import isInsideContainer from 'is-inside-container'
```
Comment (mattrosno) [inline src/main/core/environment.ts:8]: SonarCloud is flagging this line as uncovered code? @IgnacioBecerra

Comment (IgnacioBecerra) [reply src/main/core/environment.ts:8]: Unsure why this is marked as uncovered, I'm testing the function and condition itself. I'll keep looking into it..

File: README.md [hunk 3]
```diff
@@ -107,9 +108,11 @@ cryptographic function prior to storage in our database, see
...
 ### When does data get collected?
 
-Telemetry collection runs exclusively in CI environments. Collection will never happen on local
-development environments or on projects that aren't configured to run automated scripts on a CI
-environment (GitHub actions, Travis CI, etc.)
+Telemetry collection runs exclusively in CI environments. Collection will never occur on local
+development environments or on projects that aren't configured to run automated scripts in a CI
+environment (e.g., GitHub Actions, Travis CI, etc.). However, if your project runs inside a
+container (e.g., Docker, Podman), telemetry may be collected, as containerized environments are
+considered production-like and thus fall under the scope of CI workflows.
```
Comment (mattrosno) [inline README.md:115]: Minor punctuation changes here.

```suggestion
Telemetry collection runs exclusively in CI environments. Collection will never occur on local
development environments or on projects that aren't configured to run automated scripts in a CI
environment (e.g. GitHub Actions, Travis CI, etc.) However, if your project runs inside a
container (e.g. Docker, Podman), telemetry may be collected, as containerized environments are
considered production-like and thus fall under the scope of CI workflows.
```

**Review by mattrosno — CHANGES_REQUESTED** [review]
New wording looks great.

**Review by mattrosno — APPROVED** [review]

**Review by ariellalgilmore — APPROVED** [review]

**Pull request merged**